impl/maven-testing/src/main/java/org/apache/maven/api/di/testing/MavenDIExtension.java

```diff
@@ -84,18 +84,34 @@ protected void setContext(ExtensionContext context) {
      * Creates and configures the DI container for test execution.
      * Performs component discovery and sets up basic bindings.
      *
-     * @throws IllegalArgumentException if container setup fails
+     * @throws IllegalStateException if the ExtensionContext is null, the required test class is unavailable,
+     *         the required test instance is unavailable, or if container setup fails
      */
-    @SuppressWarnings("unchecked")
     protected void setupContainer() {
+        if (context == null) {
+            throw new IllegalStateException("ExtensionContext must not be null");
```

Comment:
I'm uncertain here. After thinking about this, I do think this is a proper use of IllegalStateException because it's not a null argument but rather the state of the object that's invalid. However, I might be convinced otherwise.